### PR TITLE
Handling warnings in pytest

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR suggests getting rid of use `-p warnings` flag in pytest configuration and handling warnings separately.
This will help you see new warnings in internal CI logs and during development

- [X] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/test_arraycreation.py [hunk 1]
```diff
@@ -516,6 +517,8 @@ def test_vander_seq(sequence):
     assert_allclose(vander_func(numpy, sequence), vander_func(dpnp, sequence))
 
 
+# Expected warning
+@pytest.mark.filterwarnings("ignore::numpy.ComplexWarning")
```
Comment (antonwolfy) [inline tests/test_arraycreation.py:521]: We have the existing f/w to suppress that:
```suggestion
@pytest.mark.usefixtures("suppress_complex_warning")
```

File: tests/test_arraymanipulation.py [hunk 2]
```diff
@@ -846,6 +846,8 @@ def test_asfarray(dtype, data):
     assert_array_equal(result, expected)
 
 
+# Expected warning
+@pytest.mark.filterwarnings("ignore::numpy.ComplexWarning")
```
Comment (antonwolfy) [inline tests/test_arraymanipulation.py:850]:
```suggestion
@pytest.mark.usefixtures("suppress_complex_warning")
```

File: tests/test_arraycreation.py [hunk 3]
```diff
@@ -546,7 +549,9 @@ def test_full_like(array, fill_value, dtype, order):
 
     a = numpy.array(array)
     ia = dpnp.array(array)
-    assert_array_equal(func(numpy, a), func(dpnp, ia))
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", numpy.ComplexWarning)
```
Comment (antonwolfy) [inline tests/test_arraycreation.py:553]: What behavior would be changed if to use `@pytest.mark.usefixtures("suppress_complex_warning")` instead?

**Review by antonwolfy — COMMENTED** [review]
`pytest -s` run on Gen9 reports 255 warnings:
```bash
=================================================================================================================================== warnings summary ===================================================================================================================================
tests/test_arraycreation.py: 250 warnings
  /localdisk/work/antonvol/soft/miniconda3/envs/dpnp_py39_ext/lib/python3.9/site-packages/dpctl/tensor/_type_utils.py:456: DeprecationWarning: finfo() dtype cannot be None. This behavior will raise an error in the future. (Deprecated in NumPy 1.25)
    self._finfo = np.finfo(dtype)

tests/test_sycl_queue.py::test_from_dlpack_with_dpt[complex64-level_zero:gpu:0]
  /localdisk/work/antonvol/soft/miniconda3/envs/dpnp_py39_ext/lib/python3.9/site-packages/numpy/testing/_private/utils.py:722: RuntimeWarning: invalid value encountered in equal
    func=lambda xy: xy == +inf,

tests/test_sycl_queue.py::test_from_dlpack_with_dpt[complex64-level_zero:gpu:0]
  /localdisk/work/antonvol/soft/miniconda3/envs/dpnp_py39_ext/lib/python3.9/site-packages/numpy/testing/_private/utils.py:725: RuntimeWarning: invalid value encountered in equal
    func=lambda xy: xy == -inf,

tests/third_party/cupy/core_tests/test_ndarray_math.py::TestRoundHalfway_param_0_{decimals=-3}::test_round_halfway_uint
  /localdisk/work/antonvol/soft/miniconda3/envs/dpnp_py39_ext/lib/python3.9/site-packages/dpctl/tensor/_ctors.py:646: DeprecationWarning: NumPy will stop allowing conversion of out-of-bound Python integers to integer arrays.  The conversion of 1000 to uint8 will fail in the future.
  For the old behavior, usually:
      np.array(value).astype(dtype)
  will give the desired result (the cast overflows).
    np.asarray(obj, dtype=dtype),

tests/third_party/cupy/core_tests/test_ndarray_math.py::TestRoundHalfway_param_0_{decimals=-3}::test_round_halfway_uint
  /localdisk/work/antonvol/code/dpnp/tests/third_party/cupy/core_tests/test_ndarray_math.py:108: DeprecationWarning: NumPy will stop allowing conversion of out-of-bound Python integers to integer arrays.  The conversion of 1000 to uint8 will fail in the future.
  For the old behavior, usually:
      np.array(value).astype(dtype)
  will give the desired result (the cast overflows).
    a *= xp.array(scale, dtype=dtype)

tests/third_party/cupy/creation_tests/test_ranges.py::TestRanges::test_linspace_float_underflow
  /localdisk/work/antonvol/soft/miniconda3/envs/dpnp_py39_ext/lib/python3.9/site-packages/numpy/core/function_base.py:158: RuntimeWarning: overflow encountered in multiply
    y *= step
```
Is it expected to have warning on Gen9?

File: tests/test_arraycreation.py (hunk 3 above)
Comment (vlad-perevezentsev) [reply tests/test_arraycreation.py:553]: I don't think the behavior will change. 
This is just one way of handling warnings as an option and I think using `pytest.mark.usefixtures ` looks better and cleaner

File: tests/test_arraycreation.py (hunk 1 above)
Comment (vlad-perevezentsev) [reply tests/test_arraycreation.py:521]: I overlooked it. Thanks
The warnings on Gen9 will be handled too

**Comment by antonwolfy** [comment]
There are still warnings on PVC:
```bash
=========================================================================================== warnings summary ===========================================================================================
tests/test_sycl_queue.py::test_from_dlpack_with_dpt[complex64-level_zero:gpu:0]
tests/test_sycl_queue.py::test_from_dlpack_with_dpt[complex64-level_zero:gpu:1]
  /home/gta/miniconda3/envs/dpnp_dev/lib/python3.10/site-packages/numpy/testing/_private/utils.py:722: RuntimeWarning: invalid value encountered in equal
    func=lambda xy: xy == +inf,

tests/test_sycl_queue.py::test_from_dlpack_with_dpt[complex64-level_zero:gpu:0]
tests/test_sycl_queue.py::test_from_dlpack_with_dpt[complex64-level_zero:gpu:1]
  /home/gta/miniconda3/envs/dpnp_dev/lib/python3.10/site-packages/numpy/testing/_private/utils.py:725: RuntimeWarning: invalid value encountered in equal
    func=lambda xy: xy == -inf,

```

File: tests/test_sycl_queue.py [hunk 4]
```diff
@@ -1663,6 +1663,7 @@ def test_array_creation_from_dpctl(copy, device):
     assert isinstance(result, dpnp_array)
 
 
+@pytest.mark.usefixtures("suppress_invalid_numpy_warnings")
```
Comment (antonwolfy) [inline tests/test_sycl_queue.py:1666]: Should the fixture be added to `test_from_dlpack_with_dpt` instead?

Comment (vlad-perevezentsev) [reply tests/test_sycl_queue.py]: Yes, I have fixed it.
Thank you

**Review by antonwolfy — APPROVED** [review]
Thank you @vlad-perevezentsev

**Pull request merged**